azure/table_azure_nat_gateway.go

```diff
@@ -0,0 +1,241 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION ////
+
+func tableAzureNatGateway(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_nat_gateway",
+		Description: "Azure NAT Gateways",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"name", "resource_group"}),
+			Hydrate:    getNatGateway,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isNotFoundError([]string{"ResourceNotFound", "ResourceGroupNotFound", "404"}),
+			},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listNatGateways,
+		},
+		Columns: azureColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The friendly name that identifies the nat gateway.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "Contains ID to identify a nat gateway uniquely.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "etag",
+				Description: "An unique read-only string that changes whenever the resource is updated",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "idle_timeout_in_minutes",
+				Description: "The idle timeout of the nat gateway.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromP(extractNatGatewayProperties, "idleTimeoutInMinutes"),
```

Comment:
Can we use `transform.FromField("NatGatewayPropertiesFormat.IdleTimeoutInMinutes")` here and the same for the other columns where ever applicable to fetch the properties?

Reply:
we can not use `transform.FromField` due to inconsistency in API behavior we are not getting the inner-level property properly so we must need to extract it.